src/Microsoft.AspNetCore.Mvc.Core/Routing/PageLinkGeneratorExtensions.cs

```diff
@@ -143,11 +143,11 @@ public static class PageLinkGeneratorExtensions
         public static string GetUriByPage(
             this LinkGenerator generator,
             HttpContext httpContext,
-            string page = default,
-            string handler = default,
-            object values = default,
-            string scheme = default,
-            HostString? host = default,
+            string page,
+            string handler,
+            object values,
+            string scheme,
+            HostString host,
```

Comment:
See https://github.com/aspnet/Routing/pull/833#discussion_r222560368

Reply:
Hmmm, this high-level helper method is a lot more verbose to call after this change: `_link.GetUriByPage(context, "/APage", null, null, "https", "www.contoso.com");`

Controversial idea that I don't necessarily agree with: leave the method parameters with default values
Problem with that is if it is decided to never get scheme/host from context then there is no going back.

Reply:
yeah.... I'm kinda ok with it being more verbose. It's still better than what we have today.

The kinds of features that @blowdart and I were discussing were options like the ability to use the host header iif it's been **validated** or, have a place to configure a default host to be used when it isn't provided.

Reply:
As long as the docs make it clear you should only pass a trusted host header. What happens when it's left as null? It becomes relative?

Reply:
Ok I'm closing out this PR because we're keeping the functionality. I'll send new PRs for the doc changes

@blowdart - this API generates an absolute URI using parts of the provided HttpContext - if you specify a host here, then it will be used instead of the host from the context. We also have a flavour that doesn't use HttpContext.